test_tools/src/test_tools/learn_objects.py

```diff
@@ -0,0 +1,73 @@
+#! /usr/bin/env python
+
+import sys
+
+import rospy
+import PyKDL as kdl
+
+
+from robot_smach_states.util.designators import EntityByIdDesignator
+from robot_skills.util.kdl_conversions import VectorStamped
+from robot_skills.util.robot_constructor import robot_constructor
+
+
+def look_at_entity(robot, location_des):
+    # look on top of the surface
+    entity = location_des.resolve()
+    area = "on_top_of"
+    search_volume = entity.volumes[area]
+    x_obj = 0.5 * (search_volume.min_corner.x() + search_volume.max_corner.x())
+    y_obj = 0.5 * (search_volume.min_corner.y() + search_volume.max_corner.y())
+    z_obj = search_volume.min_corner.z()
+    lookat_pos_map = entity.pose.frame * kdl.Vector(x_obj, y_obj, z_obj)
+    x = lookat_pos_map.x()
+    y = lookat_pos_map.y()
+    z = lookat_pos_map.z()
+
+    robot.head.look_at_point(VectorStamped(x, y, z, "/map"), timeout=0)
+    rospy.loginfo("Looking at position ({}, {}, {})".format(x, y, z))
+    robot.head.wait_for_motion_done()
+    rospy.loginfo("done looking a the {}".format(entity.id))
+
+
+if __name__ == '__main__':
+    rospy.init_node('learn_objects')
+
+    if len(sys.argv) < 2:
+        print("Please specify a robot name")
+        sys.exit()
+
+    robot_name = sys.argv[1]
+    if robot_name != "hero":
+        rospy.logwarn("Currently learn_objects is designed to be used with hero. It will not work for other robots")
+        sys.exit()
+
+    location = "hero_case"
+
+    robot = robot_constructor(robot_name)
+
+    location_des = EntityByIdDesignator(robot, location)
+
+    arm = robot.leftArm()
```

Comment:
I am not entirely sure what the status of the arm stuff is, but I am pretty sure that this could be generalized. Maybe @alberth or @LoyVanBeek can help with the generalisation?